grakn-core/src/main/java/ai/grakn/graql/ComputeQueryOf.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql;
+
+import ai.grakn.concept.Label;
+
+import java.util.Collection;
+
+/**
+ * yes this is a bad name TODO
+ *
+ * @param <T> the type of result this query will return
+ *
+ * @author Felix Chapman
+ */
+public interface ComputeQueryOf<T> extends ComputeQuery<T> {
+
+    /**
+     * @param resourceTypeLabels an array of types of resources to execute the query on
+     * @return a ComputeQuery with the subTypeLabels set
+     */
+    ComputeQueryOf<T> of(String... resourceTypeLabels);
+
+    /**
+     * @param resourceLabels a collection of types of resources to execute the query on
+     * @return a ComputeQuery with the subTypeLabels set
+     */
+    ComputeQueryOf<T> of(Collection<Label> resourceLabels);
+
```

Comment:
In statistics query, ofLabels must be specified, and the labels must be attribute types. In other queries, such as degree and coreness, ofLables are optional, and they can be any instance type. Maybe we should separate them? It can be difficult to explain what ofLabels are here.

Reply:
Yeah, that makes sense! Ideas for names:

- `ComputeQueryOf` -> `StatisticsQuery`
- `StatisticsQuery#ofLabels` -> `attributeLabels`
- `DegreeQuery#ofLabels` -> `targetLabels`
- `CorenessQuery#ofLabels` -> `targetLabels`

Reply:
👍